scalalib/src/mill/scalalib/JavaModule.scala

```diff
@@ -140,10 +140,45 @@ trait JavaModule
   def ivyDeps: T[Agg[Dep]] = Task { Agg.empty[Dep] }
 
   /**
-   * Aggregation of mandatoryIvyDeps and ivyDeps.
+   * Aggregation of mandatoryIvyDeps and ivyDeps, with BOMs and dependency management data
+   * added to each of them.
    * In most cases, instead of overriding this Target you want to override `ivyDeps` instead.
    */
-  def allIvyDeps: T[Agg[Dep]] = Task { ivyDeps() ++ mandatoryIvyDeps() }
+  def allIvyDeps: T[Agg[Dep]] = Task {
+    val bomDeps0 = allBomDeps().toSeq
+    val rawDeps = ivyDeps() ++ mandatoryIvyDeps()
+    val depsWithBoms =
+      if (bomDeps0.isEmpty) rawDeps
+      else rawDeps.map(dep => dep.copy(dep = dep.dep.addBoms(bomDeps0)))
+    val depMgmt = dependencyManagementDict()
+    if (depMgmt.isEmpty)
+      depsWithBoms
+    else {
+      lazy val depMgmtMap = depMgmt.toMap
+      depsWithBoms
+        .map(dep => dep.copy(dep = dep.dep.withOverrides(dep.dep.overrides ++ depMgmt)))
+        .map { dep =>
+          val key = DependencyManagement.Key(
+            dep.dep.module.organization,
+            dep.dep.module.name,
+            coursier.core.Type.jar,
+            dep.dep.publication.classifier
+          )
+          val versionOverride =
+            if (dep.dep.version == "_") depMgmtMap.get(key).map(_.version)
+            else None
+          val exclusions = depMgmtMap.get(key)
+            .map(_.minimizedExclusions)
+            .map(dep.dep.minimizedExclusions.join(_))
+            .getOrElse(dep.dep.minimizedExclusions)
+          dep.copy(
+            dep = dep.dep
+              .withVersion(versionOverride.getOrElse(dep.dep.version))
+              .withMinimizedExclusions(exclusions)
+          )
+        }
+    }
+  }
```

Comment:
This large-ish task is hard to digest. Maybe, we can make it more compact and use some `def` s with meaningful names. Or add some in-line comments.

Reply:
Maybe this logic could be moved into a helper method?

Reply:
That method isn't around any more now. What it used to do is now done in `JavaModule#{processDependency,processedIvyDeps}`.

Reply:
`processDependency` is a helper method somehow